lib/cursor_pager/page.rb

```diff
@@ -148,14 +148,19 @@ def after_limit_value
     end
 
     def limit_value_for(cursor)
-      id = encoder.decode(cursor)
-
-      selects = order_values.map(&:select_string)
-      item = ordered_relation.where(id: id).select(selects).first
+      item = limit_item_for(cursor)
 
       raise CursorNotFoundError, cursor if item.blank?
 
       order_values.map { |value| item[value.select_alias] }
     end
+
+    def limit_item_for(cursor)
+      id = encoder.decode(cursor)
+
+      selects = order_values.map(&:select_string)
+      ordered_relation.preload_values = []
+      ordered_relation.unscope(:includes).where(id: id).select(selects).first
```

Comment:
What happens when you use `eager_load` directly?

Reply:
eager_load works because it uses `LEFT OUTER JOIN` instead of additional selects

Reply:
Mind adding a test? 🙂

Reply:
I'm also a bit worried, that `ordered_relation.preload_values = []` is mutating the relation, since the same relation is also used to actually fetch the page. Seems to me like the preloads could _get lost_. Maybe we should add some tests around that? `page_records_spec.rb` would probably be a good place.

Reply:
good point, I will check that